Src/ILGPU.Algorithms/Vectors/VectorTypes.tt

```diff
@@ -451,7 +451,11 @@ namespace ILGPU.Algorithms.Vectors
         /// </summary>
         /// <returns>The readonly span instance.</returns>
         public unsafe ReadOnlySpan<<#= type.Type #>> AsSpan() =>
-            new(Unsafe.AsPointer(ref Unsafe.AsRef(this)), <#= vectorLength #>);
+#if NET8_0_OR_GREATER
+            new(Unsafe.AsPointer(ref Unsafe.AsRef(in this)), <#= vectorLength #>);
+#else
+            new(Unsafe.AsPointer(ref Unsafe.AsRef(in this)), <#= vectorLength #>);
```

Comment:
Question: should this be `Unsafe.AsRef(this)` in the `#else` case?

Reply:
Nice catch. Have pushed a fix.